src/Gateways/GpApiGateway.php

```diff
@@ -0,0 +1,119 @@
+
+use GlobalPayments\Api\Entities\Enums\Environment;
+use GlobalPayments\Api\Entities\Enums\GatewayProvider;
+use GlobalPayments\WooCommercePaymentGatewayProvider\Gateways\ThreeDSecure\CheckEnrollment;
+use GlobalPayments\WooCommercePaymentGatewayProvider\Gateways\ThreeDSecure\GpApiGateway_Webhook_Handler;
+use GlobalPayments\WooCommercePaymentGatewayProvider\Plugin;
+use WC_Order;
+
+defined( 'ABSPATH' ) || exit;
+
+class GpApiGateway extends AbstractGateway {
+    /**
+     * Defines production environment
+     */
+    const ENVIRONMENT_PRODUCTION = 'production';
+
+    /**
+     * Defines sandbox environment
+     */
+    const ENVIRONMENT_SANDBOX    = 'sandbox';
+
+    /**
+     * SDK gateway provider
+     *
+     * @var string
+     */
+    public $gateway_provider = GatewayProvider::GP_API;
+
+    /**
+	 * App ID
+	 *
+	 * @var string
+	 */
+	public $app_id;
+
+	/**
+	 * App Key
+	 *
+	 *
+	 * @var string
+	 */
+	public $app_key;
+
+	/**
+	 * Should live payments be accepted
+	 *
+	 * @var bool
+	 */
+	public $is_production;
+
+	/**
+	 * Integration's Developer ID
+	 *
+	 * @var string
+	 */
+	public $developer_id = '';
+
+	public function configure_method_settings() {
+		$this->id                 = 'globalpayments_gpapi';
+		$this->method_title       = __( 'GP-API', 'globalpayments-gateway-provider-for-woocommerce' );
+		$this->method_description = __( 'Connect to the Global Payments API (GP-API) gateway', 'globalpayments-gateway-provider-for-woocommerce' );
+	}
+
+	public function get_first_line_support_email() {
+		return 'securesubmitcert@e-hps.com';
+	}
+
+	public function get_gateway_form_fields() {
+		return array(
+            'app_id' => array(
+                'title'       => __( 'App Id', 'globalpayments-gateway-provider-for-woocommerce' ),
+                'type'        => 'text',
+                'description' => __( 'Get your App Id from your <a href="https://developer.globalpay.com/user/register" target="_blank">Global Payments Developer Account</a>.', 'wc_securesubmit' ),
+                'default'     => '',
+            ),
+            'app_key' => array(
+                'title'       => __( 'App Key', 'globalpayments-gateway-provider-for-woocommerce' ),
+                'type'        => 'text',
+                'description' => __( 'Get your App Key from your <a href="https://developer.globalpay.com/user/register" target="_blank">Global Payments Developer Account</a>.', 'wc_securesubmit' ),
+                'default'     => '',
+            ),
+			'is_production'   => array(
+				'title'   => __( 'Live Mode', 'globalpayments-gateway-provider-for-woocommerce' ),
+				'type'    => 'checkbox',
+				'default' => 'no',
+			),
+		);
+	}
+
+    public function get_frontend_gateway_options() {
+        return array(
+            'accessToken' => $this->get_access_token(),
+            'env'         => $this->is_production ? self::ENVIRONMENT_PRODUCTION : self::ENVIRONMENT_SANDBOX,
```

Comment:
I approve of this approach. Can we move these constants to AbstractGateway and leverage them within all the gateway implementations?

Reply:
Changed.